api/src/main/java/io/druid/data/input/impl/StringDimensionSchema.java

```diff
@@ -27,15 +27,19 @@
 {
   @JsonCreator
   public static StringDimensionSchema create(String name) {
-    return new StringDimensionSchema(name);
+    return new StringDimensionSchema(name, -1);
   }
 
+  private final int compareCacheSize;
+
   @JsonCreator
   public StringDimensionSchema(
-      @JsonProperty("name") String name
+      @JsonProperty("name") String name,
+      @JsonProperty("compareCacheSize") int compareCacheSize
```

Comment:
I'm not sure if this should be part of the schema, it seems to be this is mainly a runtime optimization that we should be able to enable or disable independently.